pkgs/development/libraries/libunarr/default.nix

```diff
@@ -16,11 +16,14 @@ stdenv.mkDerivation rec {
     substituteInPlace pkg-config.pc.cmake \
       --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
       --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+  '' + lib.optionalString stdenv.isDarwin ''
+    substituteInPlace CMakeLists.txt \
+      --replace '-Wl,--no-undefined -Wl,--as-needed' '-Wl,-undefined,error'
```

Comment:
Add a comment justifying why this is needed.

Reply:
Done.